### ISSUE TITLE
Loading an excel file with `read_excel(engine="calamine")` fails when a float column is all NaN's

### ISSUE THREAD

**Issue by adamporich** [pola-rs/polars#16361]
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

Sorry I can't provide a copy of the file. I can probably produce a reproduction if necessary but I believe any column that is all `NaN`'s will trigger this bug

```python
metadata_xlsx = pl.read_excel(
      b"...excelfilecontents",
      engine="calamine",
      sheet_id=0,
      raise_if_empty=False,
      infer_schema_length=None,
  )
```

### Log output

```shell
File "python_deps_polars/site-packages/polars/io/spreadsheet/functions.py", line 868, in _read_spreadsheet_calamine
    df = df.with_columns(*downcast)
  File "python_deps_polars/site-packages/polars/dataframe/frame.py", line 8310, in with_columns
    return self.lazy().with_columns(*exprs, **named_exprs).collect(_eager=True)
  File "python_deps_polars/site-packages/polars/lazyframe/frame.py", line 1816, in collect
    return wrap_df(ldf.collect(callback))
polars.exceptions.ComputeError: conversion from `f64` to `i64` failed in column 'name' for 57 out of 57 values: [NaN, NaN, … NaN]
```

### Issue description

In the `_read_spreadsheet_calamine` function the following code https://github.com/pola-rs/polars/blob/6289dc3be02347683abe9ed0f7b2deb92097872f/py-polars/polars/io/spreadsheet/functions.py#L853-L876 ends up trying to cast a column containing all `NaN`'s to `i64` and failing. I think it's the `ignore_nulls=True` in there but not terribly sure what the best fix would be.

There is a workaround of using `schema_overrides` but it seems to me that this is a bug that should be fixed

### Expected behavior

Reading a column of NaN's using `read_excel(engine="calamine")` generates a `f64` column

### Installed versions

<details>

```
--------Version info---------
Polars:               0.20.26
Index type:           UInt32
Platform:             Linux-5.15.0-102-generic-x86_64-with-glibc2.31
Python:               3.10.13 (main, Oct  3 2023, 01:22:22) [Clang 17.0.1 ]

----Optional dependencies----
adbc_driver_manager:  <not installed>
cloudpickle:          <not installed>
connectorx:           <not installed>
deltalake:            <not installed>
fastexcel:            0.10.4
fsspec:               <not installed>
gevent:               <not installed>
hvplot:               <not installed>
matplotlib:           <not installed>
nest_asyncio:         <not installed>
numpy:                1.22.4
openpyxl:             3.1.2
pandas:               1.4.4
pyarrow:              12.0.0
pydantic:             <not installed>
pyiceberg:            <not installed>
pyxlsb:               <not installed>
sqlalchemy:           <not installed>
torch:                <not installed>
xlsx2csv:             0.8.2
xlsxwriter:           <not installed>
None
```

</details>

**Comment by alexander-beedie** [comment]
Can you specify the exact error in your cells? 
Excel doesn't actually _have_ a NaN as far as I recall 😅 

Likely one of these?
```
#N/A	Not available	Can’t find the value you’re looking for
#DIV/O!	Divided by zero	The denominator is zero.
#NULL!	Empty value	Can’t determine a range of formula
#NUM!	Invalid number	Invalid numeric values
```
I generated a column full of `#NUM!` (which feels like it maps best to NaN) but got a different error: `CalamineCellError: calamine cell error: #NUM!`.

**Comment by adamporich** [comment]
:laughing: apparently in that file I have what looks like via `LibreOffice Calc` as a bunch of "nan" string values.
To be clear I'm trying to import hundreds of `.xlsx` files into Polars for export as `.csv` maybe Parquet or Arrow later on.
Attached is a screenshot from `LibreOffice Calc` with a column that generates this error I think the stacktrace was a different column with all `nan`'s this is mixed integer's and `nan`'s

**Comment by adamporich** [comment]
Still new to Polars but if I understand correctly https://docs.pola.rs/user-guide/expressions/missing-data/ then an Int64 type should be nullable. In this case it appears to be trying to convert a Float64 column to an Int64 column. In that case should the `NaN`'s be converted to `null` / `None` when it's decided to convert the column to Int64?
To be clear at least on a per file basis the conversion seems to make sense to me. Nothing was a Float and no specific schema was specified so let's convert to Int seems a decent policy

**Comment by alexander-beedie** [comment]
Hmm... We don't consider `nan` and `null` to be equivalent, so if the string "nan" was parsed here I'd expect it to return a float column with actual `nan` values in it (as NaN is a real floating point value). You could then call `df.fill_nan(None)` to change the `nan` values to `null` if that's your desired outcome 🤔 I'll have a look at the code to see how reasonable this is, heh.

**Comment by adamporich** [comment]
I believe `calamine` / `fastexcel` is correctly returning a float column. 
But as per the comment at https://github.com/pola-rs/polars/blob/6289dc3be02347683abe9ed0f7b2deb92097872f/py-polars/polars/io/spreadsheet/functions.py#L857 it looks like a column of int's and therefore we try to "cast back to int where possible." but that fails because the `NaN`'s aren't able to be cast to int's

**Comment by alexander-beedie** [comment]
> I believe `calamine` / `fastexcel` is correctly returning a float column. But as per the comment at

Confirmed... and now fixed on our side; will be in `0.20.29` ;)
Thanks for the bug report.
